synapse/util/caches/stream_change_cache.py

```diff
@@ -145,13 +157,15 @@ def has_any_entity_changed(self, stream_pos: int) -> bool:
             return True
 
     def get_all_entities_changed(self, stream_pos: int) -> Optional[List[EntityType]]:
-        """Returns all entities that have had new things since the given
+        """Returns all entities that have had changes since the given
         position. If the position is too old it will return None.
 
         Returns the entities in the order that they were changed.
         """
-        assert type(stream_pos) is int
+        assert isinstance(stream_pos, int)
 
+        # _cache is not valid before the earliest known stream position, so
+        # return that no known entities have changed.
         if stream_pos < self._earliest_known_stream_pos:
```

Comment:
I would think this should be `<=` also since `_cache` is invalid at or before that position?

This causes tests to fail.